aws/rust-runtime/aws-inlineable/src/s3_errors.rs

```diff
@@ -15,15 +17,18 @@ impl ErrorExt for smithy_types::Error {
     }
 }
 
-pub fn parse_extended_error<B>(
+pub fn parse_extended_error(
     error: smithy_types::Error,
-    response: &http::Response<B>,
+    headers: Option<&HeaderMap<HeaderValue>>,
```

Comment:
would probably just pass in an empty headermap instead of having this be optional.

Alternatively, I'd make a separate error parser function that works for event stream errors rather than trying to shove everything through this HTTP based function